assets/javascripts/kitten/components/tooltips/toggletip/index.js

```diff
@@ -338,6 +353,7 @@ Toggletip.defaultProps = {
   modifier: 'info',
   actionProps: {},
   bubbleProps: {},
+  targetElement: undefined,
```

Comment:
Est-ce ce sera assez robuste d'utiliser une condition 'falsy' sur `targetElement`, en l'initialisant à `undefined`?
`targetElement && React.isValidElement(targetElement)`
Peut-être qu'utiliser `isEmpty` ou `!!`, en initialisant à `null` pourrait être plus costaud. Qu'en penses-tu?

Reply:
Oui bonne idée !

Reply:
En même temps, le `isValidElement` devrait quand même faire le taf.

Reply:
par curiosité : pourquoi est-ce que tu utiliserais plutôt `null` qu'`undefined` ici ?

Reply:
J'aime bien cette explication sur la différence d'usage entre null et undefnined:
https://stackoverflow.com/a/5076962
J'initialise généralement mes props à `null` plutôt qu'`undefined` car je préfère me dire que mon composant n'a pas attribué de valeur aux props qu'il ne va peut-être pas recevoir.
Il y a aussi le fait qu'`undefined` n'était pas en lecture seule dans les anciennes version de JS, et qu'il peut rester des assignation bizarre dans de vieilles codebases:
https://stackoverflow.com/questions/8783510/how-dangerous-is-it-in-javascript-really-to-assume-undefined-is-not-overwritte

Donc j'ai tendance à éviter `undefined` dans mon code.

Reply:
Hmm mais quand tu sets ta defaultProp à `null`, ce n'est pas justement là que tu définies une valeur pour la prop à l'inverse de quand tu passes `undefined` ? 🤔  Pour ma part, j'ai l'habitude d'utiliser `undefined` car pour moi c'est une approche plus "neutre" : on se contente de mimiquer le comportement que l'on aurait d'office en cherchant à loguer une prop inexistante par exemple. Mais c'est pas une opinion ultra forte non plus 😉

Reply:
Ahah oui, il y a pas mal de sensibilités différentes sur cette dualité. Je n'ai pas une opinion forte dessus non plus, seulement un sentiment qu'`undefined` est moins "sûr"?
C'est aussi pour ça que j'utilise `isEmpty` souvent, ça me permet de ne pas me poser ce genre de question :)